src/sinit.c

```diff
@@ -0,0 +1,77 @@
+/* 
+ * Copyright (c) 2017, SingularityWare, LLC. All rights reserved.
+ * 
+ * This software is licensed under a 3-clause BSD license.  Please
+ * consult LICENSE file distributed with the sources of this project regarding
+ * your rights to use or distribute this software.
+ * 
+ */
+
+
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <fcntl.h>
+#include <libgen.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+#include "config.h"
+#include "util/file.h"
+#include "util/util.h"
+#include "util/daemon.h"
+#include "util/registry.h"
+#include "lib/image/image.h"
+#include "lib/runtime/runtime.h"
+#include "util/config_parser.h"
+#include "util/fork.h"
+#include "util/privilege.h"
+#include "util/suid.h"
+#include "util/sessiondir.h"
+#include "util/cleanupd.h"
+
+#include "./action-lib/include.h"
+
+#ifndef SYSCONFDIR
+#error SYSCONFDIR not defined
+#endif
+
+
+int main(int argc, char **argv) {
+    char *daemon_fd_str;
+    int daemon_fd;
+    int i;
+    
+    singularity_config_init(joinpath(SYSCONFDIR, "/singularity/singularity.conf"));
+    singularity_priv_init();
+    singularity_registry_init();
+
+    /* After this point, we are running as PID 1 inside PID NS */
```

Comment:
I don't know if this is the right file for the comment, but I noticed in testing that if I did a start followed immediately by a join, I would get an exit with 255 with an error message that the daemon wasn't found. However, then it still existed. When I added a short sleep after the start, it worked seamlessly. I think we might consider adding an internal pause to make it so that users don't hit this error.